client/src/modules/list/components/Items/ListItemDescription/ListItemDescription.jsx

```diff
@@ -0,0 +1,275 @@
+    this.setState({ isFocused: false });
+    document.removeEventListener('keydown', this.handleKeyPress);
+    this.handleDescriptionUpdate();
+  };
+
+  handleShowTextarea = () => this.setState({ isTextareaVisible: true });
+
+  handleHideTextarea = () => this.setState({ isTextareaVisible: false });
+
+  handleDescriptionVisibility = () =>
+    this.setState(({ isDescriptionVisible }) => ({
+      isDescriptionVisible: !isDescriptionVisible
+    }));
+
+  handleDescriptionUpdate = () => {
+    const {
+      descriptionInputValue: description,
+      isDescriptionUpdated
+    } = this.state;
+    const {
+      disabled,
+      itemId,
+      match: {
+        params: { id: listId }
+      },
+      name,
+      updateListItem
+    } = this.props;
+
+    if (disabled) {
+      return;
+    }
+
+    if (isDescriptionUpdated) {
+      this.setState({ pending: true });
+
+      this.pendingPromise = makeAbortablePromise(
+        updateListItem(listId, itemId, { description }, name)
+      );
+      this.pendingPromise.promise
+        .then(() => {
+          this.setState({ pending: false });
+          this.handleHideTextarea();
+        })
+        .catch(err => {
+          if (!(err instanceof AbortPromiseException)) {
+            this.setState({ pending: false });
+          }
+        });
+    } else {
+      this.handleHideTextarea();
+    }
+  };
+
+  isDescriptionUpdated = () => {
+    const { description: previousDescription } = this.props;
+    const { descriptionInputValue } = this.state;
+
+    const isDescriptionUpdated = !_isEqual(
+      _trim(previousDescription),
+      _trim(descriptionInputValue)
+    );
+
+    this.setState({ isDescriptionUpdated });
+  };
+
+  renderSaveButton = () => {
+    const { disabled } = this.props;
+    const { pending } = this.state;
+
+    return (
+      <div className="list-item-description__save-button">
+        <button
+          className="primary-button"
+          disabled={disabled || pending}
+          onClick={this.handleDescriptionUpdate}
+          type="button"
+        >
+          Save
+          {pending && (
+            <Preloader
+              size={PreloaderSize.SMALL}
+              theme={PreloaderTheme.LIGHT}
+            />
+          )}
+        </button>
+      </div>
+    );
+  };
+
+  renderDescription = () => {
+    const { description, disabled } = this.props;
+    const { pending } = this.state;
+
+    return (
+      <Linkify
+        properties={{
+          target: '_blank',
+          className: 'list-item-description__link'
+        }}
+      >
+        <p
+          className={classNames('list-item-description__description', {
+            'list-item-description__description--disabled': disabled || pending
+          })}
+        >
+          {description}
+        </p>
+      </Linkify>
+    );
+  };
+
+  renderAddButton = () => {
+    const { disabled } = this.props;
+
+    return (
+      <button
+        className="list-item-description__add-button"
```

Comment:
add class link-button to match other buttons around this area
and remove left padding